semgrep-core/src/core/Report.ml

```diff
@@ -208,7 +208,7 @@ let collate_results init_extra unzip_extra base_case_extra final_extra results =
   let matches, errors, (skipped_targets, profiling) = unzip_results results in
   {
     matches = List.flatten matches;
-    errors = List.flatten errors;
+    errors = List.flatten errors |> List.sort_uniq compare;
```

Comment:
Add a comment in `Semgrep_output_v0.atd` that we guarantee errors are sorted and deduped in core (though matches may not be), and then leave a corresponding comment here pointing users to the `Semgrep_output_v0.atd` comment

Reply:
I would also add in the comment here (`Report.ml`) that we dedup errors more aggressively to be consistent with parse errors

Reply:
Oh, one more thing---`collate_results` is run many times, since it's used by both `collate_rule_results` and `collate_pattern_results`. It's worth estimating the complexity of `List.sort_uniq` vs using a set

Reply:
In terms of that last point, do the existing docs we have for perf cover what I would need to do to test that?

Reply:
They should; tldr is `./run-benchmarks` with `ci-large-repos.yaml`. If that doesn't make any sense even after reading docs, poke me to update them